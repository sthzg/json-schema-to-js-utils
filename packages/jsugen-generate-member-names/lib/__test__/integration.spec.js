import schema from '@sthzg/jsugen-core/lib/sources/jsonSchema/testResources/example.schema';
import contactSchema from '@sthzg/jsugen-core/lib/sources/jsonSchema/testResources/example.contact';
import { generateMemberNamesModule as generate } from '../index';

describe('generate-member-names', () => {
  test('should pass full integration test', done => {
    expect.assertions(1);

    generate({ schema }).subscribe(output => {
      expect(output).toMatchSnapshot();
      done();
    });
  });

  test('should pass example.contact.json integration test', done => {
    expect.assertions(1);

    generate({ schema: contactSchema }).subscribe(output => {
      expect(output).toMatchSnapshot();
      done();
    });
  });
});
