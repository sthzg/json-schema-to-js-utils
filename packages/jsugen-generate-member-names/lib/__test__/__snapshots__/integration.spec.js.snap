// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate-member-names should pass example.contact.json integration test 1`] = `
"/**
 * @file This file is auto-generated. Do not modify its contents manually.
 */

export const ID = 'id';
export const FIRST_NAME = 'firstName';
export const LAST_NAME = 'lastName';
export const GENDER = 'gender';
export const TYPE = 'type';
export const NUMBER = 'number';
export const PHONE = 'phone';
"
`;

exports[`generate-member-names should pass full integration test 1`] = `
"/**
 * @file This file is auto-generated. Do not modify its contents manually.
 */

export const LEVEL_1_MEMBER_1 = 'level1Member1';
export const LEVEL_1_MEMBER_2 = 'level1Member2';
export const LEVEL_1_MEMBER_3 = 'level1Member3';
export const LEVEL_1_MEMBER_4 = 'level1Member4';
export const LEVEL_1_MEMBER_5 = 'level1Member5';
export const LEVEL_2_MEMBER_1 = 'level2Member1';
export const LEVEL_1_MEMBER_6 = 'level1Member6';
export const DEFINITION_MEMBER_1 = 'definitionMember1';
export const LEVEL_1_MEMBER_7 = 'level1Member7';
export const LEVEL_2_MEMBER_2 = 'level2Member2';
export const LEVEL_2_MEMBER_3 = 'level2Member3';
export const LEVEL_2_MEMBER_4 = 'level2Member4';
export const LEVEL_2_MEMBER_5 = 'level2Member5';
export const LEVEL_3_MEMBER_1 = 'level3Member1';
export const LEVEL_2_MEMBER_6 = 'level2Member6';
export const LEVEL_1_MEMBER_8 = 'level1Member8';
export const LEVEL_1_MEMBER_9 = 'level1Member9';
"
`;
